manifests/init.pp

```diff
@@ -82,6 +82,10 @@
   $renew_cron_hour                       = $letsencrypt::params::renew_cron_hour,
   $renew_cron_minute                     = $letsencrypt::params::renew_cron_minute,
   $renew_cron_monthday                   = $letsencrypt::params::renew_cron_monthday,
+  $systemd_date_spec                     = $letsencrypt::params::systemd_date_spec,
+  Enum['cron', 'systemd'] $timer_type    = $letsencrypt::params::timer_type,
+  Boolean          $manage_firewall      = $letsencrypt::params::manage_firewall,
```

Comment:
We can have several way to manage firewall settings.

The way choosen to implement is with `firewalld` service and this is RedHat like OSes specific. A possibility looks to be to rename the parameter `manage_firewalld`.

And also, what happens if i use cron as `timer_type` and enable firewalld management ?

Reply:
In first, renaming the parameter with `firewalld` (that is a specific word) instead of `firewall` (that is a generic word) should permits to know that module manage a software that is named Firewalld and not a firewall in general.

In second, Firewalld exist also with Debian like OSes, even if it is not installed by default. So it is probably expected to be installed on Debian like OSes when `manage_firewalld` is set to `true`.

In third, the module is declared compatible with BSD like OSes. I don't know if Firewalld is available with.

The firewall layer should probably not be managed in the module, it was not with cron.